### PR TITLE
Natspec

### LINKED ISSUE #80: Use natspec
Please check the status of the natspec project. https://github.com/ethereum/wiki/wiki/Ethereum-Natural-Specification-Format

### PR DESCRIPTION
**Pull request by badb**
Resolves #80

### CODE REVIEW

File: contracts/GNTAllocation.sol [hunk 1]
```diff
@@ -2,6 +2,8 @@ pragma solidity ^0.4.4;
 
 import "./Token.sol";
 
+/// @title GNT Allocation - distibution of additional tokens among developer and Golem Factory
+/// @author Golem Team
```
Comment (chfast) [inline contracts/GNTAllocation.sol]: I'm not sure we should place the author.

File: contracts/GNTAllocation.sol [hunk 2]
```diff
@@ -50,8 +52,10 @@ contract GNTAllocation {
         allocations[0xde22] =   25; //  0.25% of developers' allocations.
     }
 
-    // Allows developer to unlock its allocated tokens by transferring them back
-    // to developer's address.
+    /**
+     * @dev Allows developer to unlock its allocated tokens by transferring them back
```
Comment (chfast) [inline contracts/GNTAllocation.sol]: This should be `@notice`.

File: contracts/Token.sol [hunk 3]
```diff
@@ -54,9 +58,14 @@ contract GolemNetworkToken {
         fundingEndBlock = _fundingEndBlock;
     }
 
-    // Transfer GNT tokens from sender's account to provided account address.
-    // This function is disabled during the funding.
-    // Required state: Operational
+	/**
```
Comment (chfast) [inline contracts/Token.sol]: Can use use `///` everywhere?

File: contracts/Token.sol [hunk 4]
```diff
@@ -54,9 +58,14 @@ contract GolemNetworkToken {
...
+     * @notice Transfer GNT tokens from sender's account to provided account address.
+     * @dev This function is disabled during the funding.
```
Comment (chfast) [inline contracts/Token.sol]: Make it `@notice` as well.

File: contracts/Token.sol [hunk 5]
```diff
@@ -54,9 +58,14 @@ contract GolemNetworkToken {
...
+     * Required state: Operational
```
Comment (chfast) [inline contracts/Token.sol]: Information about states should be `@dev` or nothing.

**Comment by badb** [comment]
Mist doesn't support it, at least i can't see anything. Neither with my changes nor with natspec example contracts.

**Comment by subtly** [comment]
I understand this is a pedantic request. Can the terms "smart contract" and "contract" be removed – instead, use the terms "account code", "account" and "code"?

File: contracts/GNTAllocation.sol [hunk 6]
```diff
@@ -2,6 +2,7 @@ pragma solidity ^0.4.4;
 
 import "./Token.sol";
 
+/// @title GNT Allocation - Time-locked vault of tokens allocated to developers and Golem Factory
```
Comment (chfast) [inline contracts/GNTAllocation.sol]: Please, limit line length to 80.

File: contracts/Token.sol [hunk 7]
```diff
@@ -60,9 +62,13 @@ contract GolemNetworkToken {
         fundingEndBlock = _fundingEndBlock;
     }
 
-    // Transfer GNT tokens from sender's account to provided account address.
-    // This function is disabled during the funding.
-    // Required state: Operational
+    /// @notice Transfer `_value` GNT tokens from sender's account 
+    /// `msg.sender` to provided account address `_to`.
+    /// @dev This function is disabled during the funding.
```
Comment (chfast) [inline contracts/Token.sol]: Keep this sentence in `@notice` paragraph.

File: contracts/Token.sol [hunk 8]
```diff
@@ -104,9 +113,10 @@ contract GolemNetworkToken {
         Migrate(msg.sender, migrationAgent, _value);
     }
 
-    // Set address of migration target contract and enable migration process.
-    // Required state: Operational Normal
-    // State transition: -> Operational Migration
+    /// @dev Set address of migration target contract and enable migration process.
```
Comment (chfast) [inline contracts/Token.sol]: Use `@notice` for this sentence despite this function being available only for migration master. Please, limit line length to 80.

File: contracts/Token.sol [hunk 9]
```diff
@@ -149,11 +159,11 @@ contract GolemNetworkToken {
         Transfer(0, msg.sender, numTokens);
     }
 
-    // If cap was reached or crowdfunding has ended then:
-    // create GNT for the Golem Factory and developer,
-    // transfer ETH to the Golem Factory address.
-    // Required state: Funding Success
-    // State transition: -> Operational Normal
+    /// @dev If cap was reached or crowdfunding has ended then:
```
Comment (chfast) [inline contracts/Token.sol]: Add introducing sentence, e.g. "Finalize crowdfunding." Keep the rest in `@notice`.

File: contracts/Token.sol [hunk 10]
```diff
@@ -179,9 +189,9 @@ contract GolemNetworkToken {
         if (!golemFactory.send(this.balance)) throw;
     }
 
-    // Get back the ether sent during the funding in case the funding has not
-    // reached the minimum level.
-    // Required state: Funding Failure
+    /// @notice Get back the ether sent during the funding in case the funding has not
```
Comment (chfast) [inline contracts/Token.sol]: Please, limit line length to 80.

**Pull request merged**